.github/scripts/delete-aws-resources.sh

```diff
@@ -72,28 +69,33 @@ function delete_old_aws_resources() {
 
 RETRY_COUNT=0
 while [ $RETRY_COUNT -lt $MAX_RETRIES ]; do
-    # Trigger the function to delete the resources
-    delete_old_aws_resources
-
-    # If the function returned 0, then no resources needed to be deleted
-    # on this run. This means that all resources have been deleted.
-    if [ $? -eq 0 ]; then
-        echo "All resources deleted successfully"
-        break
-    fi
-
-    # Still have resources to delete, increase the retry count
-    RETRY_COUNT=$((RETRY_COUNT + 1))
-
-    # Check if there are more retries left
-    if [ $RETRY_COUNT -lt $MAX_RETRIES ]; then
-        # Retry after delay
-        echo "Retrying in $RETRY_DELAY seconds..."
-        sleep $RETRY_DELAY
-    fi
+  # Trigger the function to delete the resources
+  delete_old_aws_resources
+  status=$?
+
+  # If the function returned 0, then no resources needed to be deleted
+  # on this run. This means that all resources have been deleted.
+  if [ $status -eq 0 ]; then
+    echo "All resources deleted successfully"
+    break
+  fi
+
+  # Still have resources to delete, increase the retry count
+  RETRY_COUNT=$((RETRY_COUNT + 1))
+
+  # Check if there are more retries left
+  if [ $RETRY_COUNT -lt $MAX_RETRIES ]; then
+    # Retry after delay
+    sleep $RETRY_DELAY
+    echo "Retrying in $RETRY_DELAY seconds..."
+
+    # Increase the delay exponentially
+    RETRY_DELAY=$((RETRY_DELAY * 2))
```

Comment:
Moving the sleep before the retry so that we make sure to wait before each retry not after. This will make sure that the process sleeps after the first one and before the second one. So that we don't run into throttling issues on AWS side.

Reply:
I didn't understand this.. shouldn't we print `Retrying in $RETRY_DELAY seconds...` before executing sleep?

Reply:
That is right. I will swap sleep and echo commands.